pkg/controllers/user/alert/config/notifiers.go

```diff
@@ -430,3 +446,107 @@ func (c *PushoverConfig) UnmarshalYAML(unmarshal func(interface{}) error) error
 	}
 	return checkOverflow(c.XXX, "pushover config")
 }
+
+// HTTPClientConfig configures an HTTP client.
+type HTTPClientConfig struct {
```

Comment:
This and TLSConfig type and the method to generate http config from that seem to be exactly like the ones in [alertmanager](https://github.com/prometheus/alertmanager/pull/1222/files#diff-fbe65e96815b1559a386cd8521b5c79eR66) repo. Do you think we can vendor in and use them?

Reply:
Hi @mrajashree alertmanager also include other go package in that notifier and config types, like github.com/go-kit/kit, we need to vendor alertmanager, go-kit, and the package go-kit dependence, so we directly copy the go struct. If we really want to vendor alertmanager, we can update in a separate pr, this PR should be easier to test.